common/txmgr/address_state.go

```diff
@@ -0,0 +1,240 @@
+package txmgr
+
+import (
+	"sync"
+	"time"
+
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// AddressState is the state of all transactions for a given address
```

Comment:
nit: Add a more descriptive comment explaining what all this is, and does, for the reader to get an idea without looking at code.